src/components/TreeList/__stories__/TreeList.mdx

```diff
@@ -0,0 +1,315 @@
+
+### mapItemDataToProps
+
+map list item data structire to `ListItemView` [props](/docs/unstable-uselist--docs#listitemview);
+
+### containerRef
+
+The ability to pass a link to the DOM element of the container. For example, in order to control the focus of the list to activate keyboard navigation support;
+
+```tsx
+import React from 'react';
+import {type ListItemType, getItemRenderState, TreeList, Button, Alert} from '@gravity-ui/uikit';
+
+const items: ListItemType<string>[] = [
+    {data: {title: 'one'}},
+    {data: {title: 'two'}},
+    {data: {title: 'free'}},
+    {data: {title: 'four'}},
+    {data: {title: 'five'}},
+];
+
+const Component = () => {
+    const containerRef = React.useRef<HTMLDivElement>(null);
+    const listState = useListState();
+
+    const handleItemClick: TreeListOnItemClick<string> = ({
+        id,
+        disabled,
+        selected,
+        expanded,
+        groupState,
+    }) => {
+        // ...
+    };
+
+    return (
+        <>
+            <Alert
+                message="by button click. After click you can navigate list during keyboard"
+                theme="info"
+            />
+            <Button
+                onClick={() => {
+                    containerRef?.current?.focus();
+                }}
+            >
+                focus list
+            </Button>
+            <TreeList
+                items={items}
+                {...listState}
+                containerRef={containerRef}
+                onItemClick={handleItemClick}
+                mapItemDataToProps={({title}) => ({title})}
+            />
+        </>
+    );
+};
+```
+
+### getId
+
+Ability to generate an id for a list item depending on the list data. It is necessary to have access to more custom management of the state of the list. The property is optional.
+
+```tsx
+const items = [
+    {data: {id: 'id-1', title: 'some title 1'}, children: [...]},
+    {data: {id: 'id-2', title: 'some title 2'}, children: [...]},
+];
+
+<TreeList items={items} getId={({id})=> id}  />
+```
+
+### qa
+
+The ability to set a qa attribute for the container and sheet elements. The Qa attribute is also passed to the ListItemView.
+
+> Use the [getListItemQa](/docs/unstable-uselist--docs#getlistitemqa) is used to generate `qa` attributes in list items; also use this function in tests to compute a unique data attribute to access a specific list item
+
+    ```ts
+    await locator.getByTestId(getListItemQa('some-list-qa', '0')); // select the first item in the list if auto-generated IDs are not used
+    ```
+
+### className
+
+The ability to transfer custom css class name for the list container;
+
+### id
+
+The ability to set a custom id data attribute. By default, a unique identifier will be assigned;
+
+### multiple
+
+Since the state of the selected elements is controlled from above the component, this prop is necessary for the correct visual display of the selected elements;
```

Comment:
```suggestion
This prop is necessary for the correct view of the selected elements since the state of the selected elements is controlled from the parent component.
```